BSP/Simulator/DataGeneration/I2C.py

```diff
@@ -0,0 +1,62 @@
+import csv
+import os
+import fcntl
+import config
+
+TERMINATOR = ["0xff"]
+file = config.directory_path + config.files["I2C"]
+# read specific addresses out SIMILAR TO BSP_I2C_READ
+
+faultDict = {
+    "0x01" : "FAULT_HIGH_TEMP",
+    "0x02" : "FAULT_HIGH_VOLT",
+    "0x04" : "FAULT_LOW_VOLT",
+    "0x08" : "FAULT_HIGH_CURRENT",
+    "0x10" : "FAULT_WATCHDOG",
+    "0x20" : "FAULT_CAN_BUS",
+    "0x40" : "FAULT_VOLT_MISC",
+    "0x80" : "DATA_SOC"
+}
+
+"""
+@brief output the EEPROM error messages
+    Function to be called by simulate.py
+"""
+
+def EEPROM_Dump():
+    Dump = []
+    if os.stat(file).st_size != 0:
+        os.makedirs(os.path.dirname(file), exist_ok=True)
+        with open(file, 'r') as csvfile:
+            fcntl.flock(csvfile.fileno(), fcntl.LOCK_EX)    
+            csvreader = csv.reader(csvfile)
+            DataList = list(csvreader)
+            for fault in DataList:
+                if fault == TERMINATOR:
+                    break
+                # print faultDict[str(fault).strip('[]')]
+                # I'd like to printout each error's name but cannot get this dictionary printing to work
```

Comment:
So since each element in the array is actually a "["0x##"], as in each one is for some reason its OWN 1 element list, this wasn't working. I mostly resolved everything else but there's a lot of efficiency-based changes I could make if I could figure out why this is happening. I'd like to implement the dictionary-based printout before this is merged to the master but I think I'd need some help understanding why each element of the Fault[] list is itself another list.

Reply:
I think this is occurring because I2C.csv isn't a real .csv file (I think it was originally called EEPROM.txt or something like that), since each value in the file is separated by a newline character instead of a comma. Try setting the delimiter character to newline (I think this is how you would do it):
```
csvreader = csv.reader(csvfile, delimeter='\n')
```
For more information on the csv module, see:
https://docs.python.org/3/library/csv.html